src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1010,7 +1010,6 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
 
                 spentTransaction.SpendingDetails = spendingDetails;
                 spentTransaction.MerkleProof = null;
-                this.RemoveInputKeysLookupLock(spentTransaction);
```

Comment:
FYI just noticed the params in the signature `uint256 spendingTransactionId, int? spendingTransactionIndex` they are taken form a TransactionDetails class that is already found at the caller
`  if (!this.outpointLookup.TryGetValue(input.PrevOut, out TransactionData tTx))`
You could just pass that class inside to the method instead of looking it up again.

Reply:
indeed, nice one. Next PR. ;-)